src/components/split-button/split-button.interactions.stories.tsx

```diff
@@ -0,0 +1,387 @@
+import React from "react";
+import { StoryObj } from "@storybook/react";
+import { userEvent, within, expect } from "@storybook/test";
+
+import SplitButton from ".";
+import Button from "../button";
+import Box from "../box";
+
+import { allowInteractions } from "../../../.storybook/interaction-toggle/reduced-motion";
+import DefaultDecorator from "../../../.storybook/utils/default-decorator";
+import userInteractionPause from "../../../.storybook/utils/user-interaction-pause";
+
+type Story = StoryObj<typeof SplitButton>;
+
+export default {
+  title: "Split Button/Interactions",
+  component: SplitButton,
+  parameters: {
+    themeProvider: { chromatic: { theme: "sage" } },
+  },
+};
+
+const Wrapper = (props: { children: React.ReactNode }) => (
+  <Box display="flex" flexDirection="column" gap="32px">
+    {props.children}
+  </Box>
+);
+
+export const ButtonTypes: Story = {
+  render: () => (
+    <Wrapper>
+      {(["primary", "secondary"] as const).map((buttonType) => (
+        <Box key={buttonType} mb={3}>
+          <SplitButton
+            buttonType={buttonType}
+            text={`Split button - ${buttonType}`}
+          >
+            <Button>Button 1</Button>
+            <Button>Button 2</Button>
+            <Button>Button 3</Button>
+          </SplitButton>
+        </Box>
+      ))}
+      <Box p={2} width="298px" backgroundColor="#000">
+        <SplitButton
+          buttonType="secondary"
+          text="Split button - secondary - white"
+          isWhite
+        >
+          <Button>Button 1</Button>
+          <Button>Button 2</Button>
+          <Button>Button 3</Button>
+        </SplitButton>
+      </Box>
+    </Wrapper>
+  ),
+  play: async ({ canvasElement }) => {
+    if (!allowInteractions()) return;
+    const canvas = within(canvasElement);
+    const toggleButtons = canvas.getAllByRole("button", { name: /show more/i });
+
+    for (const toggle of toggleButtons) {
+      await userEvent.click(toggle);
+      await userInteractionPause(500);
+      await userEvent.keyboard("{Escape}");
```

Comment:
question: doesn't pressing `Escape` here close the popover? We want to capture that in chromatic

Reply:
@DobroTora all but one of the tests still calls escape and so we're taking snapshots of the component closed